components/terrain/viewdata.cpp

```diff
@@ -162,11 +162,6 @@ ViewData *ViewDataMap::getViewData(osg::Object *viewer, const osg::Vec3f& viewPo
             vd->copyFrom(*mostSuitableView);
             return vd;
         }
-        else if (!mostSuitableView)
```

Comment:
With this change QuadTreeWorld becomes allowed to reuse chunks from different distances instead of building approproate ones. This causes grass pop-ins, when QuadTreeWorld decides to use an empty chunk (from distance beyound rendering distance) for nearby objects instead of building a chunk with data.

Reply:
The issue you experienced is the logical consequence when there is no preloading in place as the code copying across views assumed. If we do not experience this anymore the fix is not necessary if not counterproductive.

Reply:
Well, with this PR this issue exists and pop-ints actually happen.

Reply:
Then it seems we have other issues the mentioned change hides.

In theory, you should only see pop-in when the preloader is too slow. In this case, would you have rather have pop-in or a loading break?

In practice, the preloader uses a predicted position 1 second in the future. It is usually not the real position at the time of loading. In my experience, the preloader is much faster and the 1 second value is too high. The time should be adjusted by the actual loading time or there should be a distance threshold to mitigate the inaccuracy. Maybe both. I will try to add a threshold, see if this is enough. The impact of this additional loading distance is minor as Any old name explains, the vertices discarded by the shader have little impact.

Reply:
> In theory, you should only see pop-in when the preloader is too slow. In this case, would you have rather have pop-in or a loading break?

If I understood correctly, it does not work like that. QuadTreeWorld has zero reasons to render new chunk  immediately as it becomes loaded. It will be rendered only when it will be requested in the next time (e.g. during cell change).

Reply:
No there is a reason. We call getViewData every frame https://github.com/OpenMW/openmw/blob/44d1a77a875a6211ae51809cb43fd0d251a5cba7/components/terrain/quadtreeworld.cpp#L436 which copies in the most appropriate view available (usually one made by the preloader) if the current view is beyond reusedistance.
https://github.com/bosvensson1/openmw/blob/726c2ac7adbcc64dfed507f729205b48a05efcae/components/terrain/viewdata.cpp#L144

At the moment though the check for rendering distance does not account for reusedistance. Will try to add that.